tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/PostResourceIntTest.java

```diff
@@ -397,6 +399,164 @@ public void shouldAllowMMultipleOtherSpecialties() throws Exception {
+        .andExpect(jsonPath("$.fieldErrors[0].field").value("specialties"));
+
+    Post postAfterUpdate = postRepository.findById(post.getId()).orElse(null);
+    Set<PostSpecialty> updatedPostSpecialtySet = postAfterUpdate.getSpecialties();
+    List <PostSpecialty> addedSubspecialties = updatedPostSpecialtySet.stream()
+        .filter(ps -> ps.getPostSpecialtyType().equals(PostSpecialtyType.SUB_SPECIALTY))
+        .collect(Collectors.toList());
+
+    assertThat(addedSubspecialties.size()).isZero();
+    assertThat(postAfterUpdate).isEqualTo(post);
+  }
+
+  @Test
+  @Transactional
+  public void shouldAllowUpdateOfPostIfSubspecialtyIsNotASubspecialty() throws Exception {
+    // initialize database with a Post
+    // the Post needs a Primary specialty as well, otherwise it wouldn't be possible to set a
+    // sub_specialty
+    post = createEntity();
+    PostSpecialty primaryPostSpecialty = createPostSpecialty(specialty, PostSpecialtyType.PRIMARY, post);
+    post.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty)));
+    postRepository.saveAndFlush(post);
+
+    // Attempt to update a Post that has a specialty of specialtyType SUB_SPECIALTY as a subspecialty
+    // (i.e. link Post and a Specialty that is SpecialtyType.SUB_SPECIALTY in a PostSpecialty in
+    // a PostSpecialty of PostSpecialtyType.SUB_SPECIALTY)
+    // The update should succeed.
+
+    Post updatedPost = createEntity();
+    updatedPost.setId(post.getId());
+
+    Specialty aSubspecialty = createSpecialty();
+    aSubspecialty.setSpecialtyTypes(new HashSet<>(
+        Collections.singletonList(SpecialtyType.SUB_SPECIALTY)));
+    em.persist(aSubspecialty);
+
+    PostSpecialty subspecialtyPostSpecialty = createPostSpecialty(aSubspecialty,
+        PostSpecialtyType.SUB_SPECIALTY, post);
+    post.getSpecialties().stream().findFirst().ifPresent(ps -> primaryPostSpecialty.setId(ps.getId()));
+    updatedPost.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty, subspecialtyPostSpecialty)));
+
+    PostDTO updatedPostDto = postMapper.postToPostDTO(updatedPost);
+
+    restPostMockMvc.perform(put("/api/posts")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(TestUtil.convertObjectToJsonBytes(updatedPostDto)))
+        .andExpect(status().isOk());
+
+    Post resultingUpdatedPost = postRepository.findById(post.getId()).orElse(null);
+    Set<PostSpecialty> updatedPostSpecialtySet = resultingUpdatedPost.getSpecialties();
+    List <PostSpecialty> addedSubspecialties = updatedPostSpecialtySet.stream()
+        .filter(ps -> ps.getPostSpecialtyType().equals(PostSpecialtyType.SUB_SPECIALTY))
+        .collect(Collectors.toList());
+
+    assertThat(addedSubspecialties.size()).isEqualTo(1);
+    assertThat(addedSubspecialties.get(0).getSpecialty()).isEqualTo(aSubspecialty);
+  }
+
+  @Test
+  @Transactional
+  public void shouldFailToCreatePostIfSubspecialtyIsNotASubspecialty() throws Exception {
```

Comment:
Maybe rename to ...IsNotOfTypeSubspecialty' for clarity

Reply:
same as above